pkg/operator/observe_config.go

```diff
@@ -92,12 +84,42 @@ func (c ConfigObserver) sync() error {
 	// don't worry about errors
 	currentConfig := map[string]interface{}{}
 	json.NewDecoder(bytes.NewBuffer(operatorConfig.Spec.ObservedConfig.Raw)).Decode(&currentConfig)
+	observedConfig := map[string]interface{}{}
+	json.NewDecoder(bytes.NewBuffer(operatorConfig.Spec.ObservedConfig.Raw)).Decode(&observedConfig)
+
+	var errors []error
+	for _, observer := range c.observers {
+		var e []error
+		observedConfig, e = observer(c.kubeClient, &rest.Config{}, observedConfig)
+		errors = append(errors, e...)
+	}
 	if reflect.DeepEqual(currentConfig, observedConfig) {
-		return nil
+		glog.Infof("writing updated observedConfig: %v", diff.ObjectDiff(operatorConfig.Spec.ObservedConfig.Object, observedConfig))
+		operatorConfig.Spec.ObservedConfig = runtime.RawExtension{Object: &unstructured.Unstructured{Object: observedConfig}}
+	}
+	if len(errors) > 0 {
+		var message bytes.Buffer
+		var i int
+		for i, err = range errors {
+			if i > 0 {
+				message.WriteString("\n")
+			}
+			message.WriteString(err.Error())
+		}
+		v1alpha1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, v1alpha1.OperatorCondition{
+			Type:    "Failing",
+			Status:  v1alpha1.ConditionTrue,
+			Reason:  "SynchronizationError",
```

Comment:
Start a file with our special reasons.  Also this name is bad.  `ConfigObservationError`

Reply:
A file in source? Or a shared document?